File: frontend/src/store/api/result/dto/product/variant/MicrophonesVariantOutputDTO.ts
import { ProductVariantOutputDTO } from "./ProductVariantOutputDTO";

export interface MicrophonesVariantOutputDTO extends ProductVariantOutputDTO {
    color: string,
    microphoneId: number
}

export function isMicrophoneVariantOutputDTO(variant: ProductVariantOutputDTO): variant is MicrophonesVariantOutputDTO {
    return "microphoneId" in variant;
}

export function isMicrophoneVariantOutputDTOArray(variants: ProductVariantOutputDTO[]): variants is MicrophonesVariantOutputDTO[] {
    return variants.every(isMicrophoneVariantOutputDTO);
}

File: backend/product-client/src/main/java/org/andante/microphones/dto/MicrophonesVariantOutputDTO.java
package org.andante.microphones.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.andante.product.dto.ProductVariantOutputDTO;

@SuperBuilder
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class MicrophonesVariantOutputDTO extends ProductVariantOutputDTO {

    private String color;
    private Long microphoneId;
}
